src/serializer/ResidualHeapVisitor.js

```diff
@@ -276,7 +276,12 @@ export class ResidualHeapVisitor {
   visitValueArray(val: ObjectValue): void {
     this.visitObjectProperties(val);
     const realm = this.realm;
-    let lenProperty = Get(realm, val, "length");
+    let lenProperty;
+    if (val.isLeakedObject()) {
+      lenProperty = this.realm.evaluateWithoutLeakLogic(() => Get(realm, val, "length"));
```

Comment:
Is it important to be able to evaluate things with leak logic after interpretation is over? Otherwise you could just set `ignoreLeakLogic` true before visiting + serialization.

Reply:
That's the part I'm not sure about. My impression was that indeed it's not important: https://github.com/facebook/prepack/pull/1382#discussion_r165111775. But maybe I'm wrong. For now I'd keep this targeted fix, and double check with @sebmarkbage when he's more available.

Reply:
Sent this as a followup https://github.com/facebook/prepack/pull/1384